### PR TITLE
Sonar Cleanup and JsonProperty Fixes

### PR DESCRIPTION
**Pull request by khanjal**
This pull request introduces a new JSON serialization approach for entities with the `ColumnAttribute`, allowing property names to be customized for JSON output and input. It adds comprehensive unit tests to verify this behavior, refactors the entity sheet mapping validation logic for better readability, and removes a legacy formula comparison test.

**Serialization improvements:**

* Added `ColumnAttributeNamingPolicy`, `ColumnAttributeJsonConverterFactory`, and `ColumnAttributeJsonConverter<T>` in `RaptorSheets.Core/Serialization/ColumnAttributeNamingPolicy.cs` to enable JSON serialization/deserialization using `ColumnAttribute.JsonPropertyName`, with fallback to camelCase when not specified.
* Introduced `JsonSerializerOptionsExtensions` in `RaptorSheets.Core/Serialization/JsonSerializerOptionsExtensions.cs` to easily configure `JsonSerializerOptions` to use the new column attribute naming policy.

**Testing enhancements:**

* Added `ColumnAttributeJsonSerializationTests` in `RaptorSheets.Core.Tests/Unit/Serialization/ColumnAttributeJsonSerializationTests.cs` to thoroughly test serialization and deserialization of entities with various `ColumnAttribute` configurations, including edge cases.
* Updated the test project file to include the new `Unit\Serialization` folder.

**Validation logic refactor:**

* Refactored `ValidateEntitySheetMapping<T>` in `EntitySheetOrderHelper.cs` to extract sheet name, order, and uniqueness validation into separate methods, improving clarity and maintainability.

**Test cleanup:**

* Removed the legacy formula comparison test from `FormulaBuilderIntegrationTests.cs` as part of test suite modernization.

### CODE REVIEW

File: RaptorSheets.Gig.Tests/RaptorSheets.Gig.Tests.csproj [hunk 1]
```diff
@@ -89,4 +89,8 @@
     <ProjectReference Include="..\RaptorSheets.Test\RaptorSheets.Test.Common.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Unit\Serialization\" />
+  </ItemGroup>
```
[the remark this answers is not in the document]
Comment (khanjal) [reply RaptorSheets.Gig.Tests/RaptorSheets.Gig.Tests.csproj]: Removed it
